packages/cli/src/zosfiles/compare/lfuss/LocalfileUss.handler.ts

```diff
@@ -0,0 +1,128 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { AbstractSession, IHandlerParameters, ImperativeError, ITaskWithStatus, TaskStage, DiffUtils } from "@zowe/imperative";
+import { Get, IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+import { ZosFilesBaseHandler } from "../../ZosFilesBase.handler";
+
+/**
+ * Handler to compare the local file and uss file's content
+ * @export
+ */
+export default class LocalfileUssHandler extends ZosFilesBaseHandler {
+    public async processWithSession(commandParameters: IHandlerParameters, session: AbstractSession): Promise<IZosFilesResponse> {
+        const task: ITaskWithStatus = {
+            percentComplete: 0,
+            statusMessage: "Retrieving local file",
+            stageName: TaskStage.IN_PROGRESS
+        };
+
+        commandParameters.response.progress.startBar({ task });
+
+        let localFile: string;
+
+        // resolving to full path if local path passed is not absolute
+        if (path.isAbsolute(commandParameters.arguments.localFilePath)) {
+            localFile = commandParameters.arguments.localFilePath;
+        } else {
+            localFile = path.resolve(commandParameters.arguments.localFilePath);
+        }
+
+        // check if the path given is of a file or not
+        fs.lstat(localFile, (err, stats) => {
+            if (err == null && stats.isFile()) {
+                return true;
+            } else if (err == null) {
+                throw new ImperativeError({
+                    msg: 'Path given is not a file'
+                });
+            } else {
+                throw new ImperativeError({
+                    msg: 'File not found, please recheck your file path'
+                });
+            }
+        });
```

Comment:
Currently, there is no guarantee the callback will get executed before the subsequent lines of code execute.

I would suggest refactoring this code to use either the synchronous method `fs.lstatSync` or the promisified method `fs.lstat` from the "fs/promises" module.

Reply:
Hey @t1m0thyj 
Refer:  https://github.com/zowe/zowe-cli/pull/1478/commits/0bcde3a5d195f85be348efbe704994df05c1a1e3

This has been resolved now :yum:

Reply:
Thank you 👍